fix(otelgin): consider `OTEL_SEMCONV_STABILITY_OPT_IN` for trace attributes (#7215)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

  - `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp`
- Cleaned up indentations under `Unreleased/Fixed` of `./CHANGELOG.md`. (#7163)
- Removed a duplicate instance of the `Changed` subheader under `1.18.0/0.43.0/0.12.0` in `./CHANGELOG.md`. (#7163)

> - Fixed in `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin` by considering the `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable for trace attributes. (#7215)

Reviewer: ```suggestion
- Fixed handling of the `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable in `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin`. (#7215)
```
